pkg/asset/installconfig/openstack/validvaluesfetcher.go

```diff
@@ -28,9 +28,7 @@ func getCloudNames() ([]string, error) {
 // getExternalNetworkNames interrogates OpenStack to get the external network
 // names.
 func getExternalNetworkNames(cloud string) ([]string, error) {
-	conn, err := clientconfig.NewServiceClient("network", &clientconfig.ClientOpts{
-		Cloud: cloud,
-	})
+	conn, err := clientconfig.NewServiceClient("network", defaultValidValuesFetcherClientOpts(cloud))
```

Comment:
Related, but not necessarily in this patch, I would refactor all of these functions to take a ProviderClient instead of a cloud name. The reason is you pull the code managing authentication options into a single place, and also you only hit keystone once rather than every time you create a new service client.